src/hotspot/share/utilities/vmError.cpp

```diff
@@ -347,14 +347,14 @@ static frame next_frame(frame fr, Thread* t) {
   }
 }
 
-void VMError::print_native_stack(outputStream* st, frame fr, Thread* t, bool print_source_info, char* buf, int buf_size) {
+void VMError::print_native_stack(outputStream* st, frame fr, Thread* t, bool print_source_info, int max_frames, char* buf, int buf_size) {
 
   // see if it's a valid frame
   if (fr.pc()) {
     st->print_cr("Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)");
-
+    const int limit = max_frames == -1 ? StackPrintLimit : max_frames;
```

Comment:
Only thing is that we may now print more than StackPrintLimit. Maybe clamping max_frames is more appropriate. 

But I do not see a reason why someone would lower StackPrintLimit and use ErrorLogSecondaryErrorDetails.